scripts/enableTrading.ts

```diff
@@ -0,0 +1,184 @@
+import Contracts from '../components/Contracts';
+import { DeployedContracts, execute, getNamedSigners, InstanceName } from '../utils/Deploy';
+import Logger from '../utils/Logger';
+import { DEFAULT_DECIMALS, TokenSymbol } from '../utils/TokenData';
+import '@nomiclabs/hardhat-ethers';
+import '@typechain/hardhat';
+import { CoinGeckoClient } from 'coingecko-api-v3';
+import Decimal from 'decimal.js';
+
+interface EnvOptions {
+    ENABLE_TRADING?: boolean;
+}
+
+const { ENABLE_TRADING: enableTrading }: EnvOptions = process.env as any as EnvOptions;
+
+interface TokenData {
+    address: string;
+    ethVirtualBalance: Decimal;
+    tokenVirtualBalance: Decimal;
+}
+
+const MAX_PRECISION = 16;
+
+const main = async () => {
+    const { deployer } = await getNamedSigners();
+    const carbonPOL = await DeployedContracts.CarbonPOL.deployed();
+
+    const tokenAddressesEnv = process.env.TOKEN_ADDRESSES;
+    if (!tokenAddressesEnv) {
+        console.error(`no tokens passed in - pass in token addresses like so: TOKEN_ADDRESSES='0x...','0x...'`);
+        return;
+    }
+
+    // Remove single quotes and whitespace, then split by commas
+    const allTokens = tokenAddressesEnv
+        .replace(/'/g, '')
+        .split(',')
+        .map((address) => address.trim());
+
+    const client = new CoinGeckoClient({
+        timeout: 10000,
+        autoRetry: true
+    });
+
+    /* eslint-disable camelcase */
+    const tokenPrices = {
+        ...(await client.simpleTokenPrice({
+            id: 'ethereum',
+            contract_addresses: [...allTokens].join(','),
+            vs_currencies: 'USD'
+        }))
+    };
+    /* eslint-enable camelcase */
+
+    const ethPriceRes = await client.simplePrice({
+        ids: 'ethereum',
+        vs_currencies: 'USD'
+    });
+
+    const ethPrice = new Decimal(ethPriceRes.ethereum.usd);
+
+    Logger.log();
+    Logger.log('Looking for disabled tokens...');
+
+    const unknownTokens: Record<string, string> = {};
+
+    const tokens: Record<string, TokenData> = {};
+
+    let symbol: string;
+    let decimals: number;
+    for (let i = 0; i < allTokens.length; i++) {
+        const token = allTokens[i];
+
+        const tokenContract = await Contracts.ERC20.attach(token, deployer);
+        symbol = await tokenContract.symbol();
+        decimals = await tokenContract.decimals();
```

Comment:
Note that this will fail for a few tokens (i.e., an exception will be thrown), but we can cross this bridge when we'd get there...

Reply:
added support for it